lib/content_publisher/featured_document_migrator.rb

```diff
@@ -0,0 +1,32 @@
+module ContentPublisher
+  class FeaturedDocumentMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
+
+      public_url = Whitehall.url_maker.public_document_url(edition)
+      public_updated_at = (edition.public_timestamp || edition.updated_at)
+
+      features = Feature.includes(:feature_list).where(document_id: document.id)
+
+      features.each do |feature|
+        offsite_link = OffsiteLink.create(title: edition.title,
+                                          summary: edition.summary,
+                                          link_type: "content_publisher_#{link_type(edition)}",
+                                          url: public_url,
+                                          date: public_updated_at,
+                                          parent: feature.feature_list.featurable)
+        feature.update!(document_id: nil, offsite_link_id: offsite_link.id)
```

Comment:
Should this be `document: nil, offsite_link: offsite_link` rather than individual ids? - I don't mind too much so long as it's consistent with the parent call.